airbyte-integrations/connectors/source-pinterest/source_pinterest/spec.json

```diff
@@ -11,6 +11,9 @@
         "type": "string",
         "title": "Start Date",
         "description": "A date in the format YYYY-MM-DD. If you have not set a date, it would be defaulted to latest allowed date by api (89 days from today).",
+        "format": "date",
```

Comment:
It look like `String does not match format YYYY-MM-DD` we get during check. 
Let's catch error and rise `AirbyteTracedException` with `config_error` failure type.

Reply:
I think it might be bets to do both! 
* Adding the pattern to the spec makes it so that users can only enter the right pattern in the UI, so they'll never run into it in check 
* Excepting the issue in check and emitting a config_error or failed connection_status message will make it so that anyone using the docker interface will know its not a system_error

Reply:
updated